doc/engineering/coding_standards.md

```diff
@@ -0,0 +1,236 @@
+# Code Development Guidelines
+
+Expectations for 2600Hz engineers as well as folks contributing to the platform.
+
+## Coding Standards
+
+The most important rule: **Be consistent both with your code and the project.**
+
+If, in the project, we have been using a variable name to represent an object, continue that in your code. Attempt to mirror existing conventions first, even if you may not agree with them! We can discuss changing it globally but every effort should be made to have a unified and consistent code base.
+
+The full coding standard that we tend to use, and will likely adapt shortly in our documentation effort, is available [here](https://github.com/inaka/erlang_guidelines).
+
+Some of the highlights:
+
+* Loud errors
+  * Don't let errors and exceptions go unlogged. Even when you handle them, write a logline with the stack trace.
+  * If it is a log line that requires manual intervention, or action, then it should should be documented in “runbook” style for a system operator.
+* Keep functions small
+  * Try to write functions with a small number of expressions, and that do only one thing. 12 expressions per function except for integration tests is a good measure.
+* Maintain existing style
+  * When editing a module written by someone else, stick to the style in which it was written. If a project has an overall style, stick to that when writing new modules as well.
+* Use your spacebar
+  * Surround operators and commas with spaces.
+* No Trailing Whitespace
+  * Remove trailing whitespace from your lines
+* Avoid deep nesting
+  * Avoid nesting case or any conditionals.  If nesting is required do not exceed three levels.  It is better to call a well-named function to encapsulate the decision making of the inner case.
+* Do not use if statements.
+  * In some circumstances if introduces static boolean logic in your code, reducing code flexibility. In other cases, a case or a function call with pattern matching in its clauses is just more declarative. For newcomers (that have learned to use if in other languages), Erlang's if can be either hard to understand or easily abused.
```

Comment:
`if`s are dangerous because `,;` typos happen and go unnoticed, and it accepts more than just guards. As always, carefulness prevails…